### PR TITLE
Fix trailing slash bug in VMR synchronization

### PR DESCRIPTION
**Pull request by MilenaHristova**
https://github.com/dotnet/arcade/issues/10799

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/arcade-services?pullRequest=2051) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/arcade-services?pullRequest=2051) before we can accept your contribution.<br/><br/>:x: MilenaHristova [sign now](https://cla.dotnetfoundation.org/dotnet/arcade-services?pullRequest=2051)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/arcade-services?pullRequest=2051) it.</sub>

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs [hunk 1]
```diff
@@ -576,7 +577,7 @@ public async Task CreatePatchesWithSubmoduleUrlChangedTest()
         expectedArgs = GetExpectedGitDiffArguments(
             expectedSubmodulePatchName1, SubmoduleSha1, Constants.EmptyGitObject, null)
             .Take(7)
-            .Append(":(glob,attr:!vmr-ignore)**/*")
+            .Append(":(glob,attr:!vmr-ignore)**")
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs:580]: Why did you remove this? I don't think this is right

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs [hunk 2]
```diff
@@ -585,11 +586,11 @@ public async Task CreatePatchesWithSubmoduleUrlChangedTest()
                 expectedArgs,
                 It.IsAny<CancellationToken>()),
                 Times.Once);
-        
+
         expectedArgs = GetExpectedGitDiffArguments(
             expectedSubmodulePatchName2, Constants.EmptyGitObject, SubmoduleSha2, null)
             .Take(7)
-            .Append(":(glob,attr:!vmr-ignore)**/*")
+            .Append(":(glob,attr:!vmr-ignore)**")
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs:593]: Same here

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs [hunk 3]
```diff
@@ -647,7 +695,7 @@ private void VerifyGitCall(string[] expectedArguments, string repoDir = VmrPath,
             ":(glob,attr:!vmr-ignore)src/*",
             ":(exclude,glob,attr:!vmr-preserve)*.dll",
             ":(exclude,glob,attr:!vmr-preserve)*.exe",
-            ":(exclude,glob,attr:!vmr-preserve)src/**/tests/**/*.*",
+            ":(exclude,glob,attr:!vmr-preserve)src/*tests/**.*",
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs:698]:
```suggestion
            ":(exclude,glob,attr:!vmr-preserve)src/**/tests/**/*.*",
```

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs [hunk 4]
```diff
@@ -615,6 +616,53 @@ public async Task CreatePatchesWithSubmoduleUrlChangedTest()
         });
     }
 
+    [Test]
+    public async Task PatchIsAppliedOnRepoWithTrailingSlashTest()
+    {
+        //Setup
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs:622]:
```suggestion
        // Setup
```

**Review by premun — CHANGES_REQUESTED** [review]

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs [hunk 5]
```diff
@@ -615,6 +616,53 @@ public async Task CreatePatchesWithSubmoduleUrlChangedTest()
...
+        {
+            "apply",
+            "--cached",
+            "--ignore-space-change",
+            "--directory",
+            $"src/{IndividualRepoName}/",
+            patch.Path,
+        },
+        VmrPath + "/"
+        );
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.DarcLib.Tests/VirtualMonoRepo/VmrPatchHandlerTests.cs:656]: Just a styling nitty comment:

```suggestion
        VmrPath + "/");
```

File: src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPatchHandler.cs [hunk 6]
```diff
@@ -234,8 +234,12 @@ public async Task ApplyPatch(SourceMapping mapping, VmrIngestionPatch patch, Can
...
         var destPath = _vmrInfo.GetRepoSourcesPath(mapping)
             .Replace(_vmrInfo.VmrPath, null)
-            .Replace("\\", "/")
-            [1..];
+            .Replace("\\", "/");
+
+        if (destPath[0] == '/')
+        {
+            destPath = destPath.Substring(1);
+        }
```
Comment (premun) [inline src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPatchHandler.cs:242]: I think you can also do:

```suggestion
            .Replace("\\", "/")
            .TrimStart('/');
```

**Review by premun — APPROVED** [review]
Looks great now!
